src/Command/VotingPlatformConfigureCommand.php

```diff
@@ -0,0 +1,237 @@
+
+        $designations = $this->designationRepository->getIncomingDesignations($date);
+
+        $this->io->progressStart();
+
+        foreach ($designations as $designation) {
+            if (DesignationTypeEnum::COMMITTEE_ADHERENT === $designation->getType()) {
+                $this->configureCommitteeElections($designation);
+            } else {
+                $this->io->error(sprintf('Unhandled designation type "%s"', $designation->getType()));
+            }
+        }
+
+        $this->io->progressFinish();
+    }
+
+    private function configureCommitteeElections(Designation $designation): void
+    {
+        $offset = 0;
+
+        while ($committeeElections = $this->committeeElectionRepository->findAllByDesignation($designation, $offset)) {
+            foreach ($committeeElections as $committeeElection) {
+                $committee = $committeeElection->getCommittee();
+
+                $this->io->progressAdvance();
+
+                if ($this->electionRepository->hasElectionForCommittee($committee, $designation)) {
+                    continue;
+                }
+
+                if (!$this->isValidCommitteeElection($committeeElection)) {
+                    continue;
+                }
+
+                $this->configureNewElection(new Election($designation), new ElectionEntity($committee));
+            }
+
+            $this->entityManager->clear();
+
+            $designation = $this->entityManager->merge($designation);
+
+            $offset += \count($committeeElections);
+        }
+    }
+
+    private function configureNewElection(Election $election, ElectionEntity $electionEntity): void
+    {
+        $election->setElectionEntity($electionEntity);
+        $committee = $electionEntity->getCommittee();
+
+        // Create candidates groups
+        $candidacies = $this->committeeCandidacyRepository->findByCommittee($committee);
+
+        foreach ($candidacies as $candidacy) {
+            $adherent = $candidacy->getCommitteeMembership()->getAdherent();
+
+            $group = new CandidateGroup();
+            $group->addCandidate($candidate = new Candidate(
+                $adherent->getFirstName(),
+                $adherent->getLastName(),
+                $candidacy->getGender()
+            ));
+
+            $candidate->setImagePath($candidacy->getImagePath());
+            $candidate->setBiography($candidacy->getBiography());
+
+            $election->addCandidateGroup($group);
+        }
+
+        $memberships = $this->committeeMembershipRepository->findVotingMemberships($committee);
+
+        $list = new VotersList($election);
+
+        foreach ($memberships as $membership) {
+            $adherent = $membership->getAdherent();
+
+            $list->addVoter($this->voterRepository->findForAdherent($adherent) ?? new Voter($adherent));
+        }
+
+        $this->entityManager->persist($list);
+        $this->entityManager->persist($election);
+        $this->entityManager->flush();
```

Comment:
maybe you should flush just before the clear in the `configureCommitteeElections` method ?

Reply:
here I prefer flush after each committee instead of doing it by batch. If the process fails, it fails for one committee and not for many committees.